feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -316,12 +316,12 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			// for devices that return the nexthop without resolving it recursively. For a->b->c the device returns b
 			// note that b->c is validated in the next code block
 			if got != ateIndirectNH {
+				if nh.GetInterfaceRef().GetInterface() == "" {
```

Comment:
This whole block of code (starting from `if got := nh.GetIpAddress(); got != atePort2.IPv4`) lacks of readability and confuses ppl.  Let's take this chance to improve it. 

How about something like below? 
```
if got := nh.GetIpAddress(); got != atePort2.IPv4 && got != ateIndirectNH {
	t.Errorf("next-hop is incorrect: got %v, want %v or %v ", got, ateIndirectNH, atePort2.IPv4)
}
```